### PR TITLE
Create windows build distribution bundle in GHA

### PR DESCRIPTION
**Pull request by akrieger**
#### Summary
None

#### Purpose of change
The MSX cross compiler was limiting our ability to use c++14 functionality like constexpr in some places. Long term it's a blocker to moving to c++17 because there isn't a newer prebuilt available and building a new version on the fly in every build is not palatable. As a result, it's beneficial to actually build our Windows builds on Windows machines from the VS projects themselves.

#### Describe the solution
Update `release.yml` to build the Windows builds on Windows workers using the same basic framework as the newly created Windows PR builds. Add a helper powershell script to assemble the distribution into a zip so we can make changes to the distribution without changing the actual build release file.

#### Describe alternatives you've considered
- Keep using MSX and try to update it (nah).
- Cross compile using clang, which would've necessitated setting up a full cross compile toolchain. Too much like my day job.
- Could've broken up `release.yml` into sub-actions. It might still be hygienic to do so in the future.

#### Testing
https://github.com/akrieger/Cataclysm-DDA/releases/tag/cdda-experimental-2021-09-22-0546 has valid builds.

#### Additional context

Ping @jbytheway.

### CODE REVIEW

File: build-scripts/windist.ps1 [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+cp Cataclysm-vcpkg-static-Release-x64.exe bindist/cataclysm-tiles.exe
+cp tools/format/JsonFormatter-vcpkg-static-Release-x64.exe bindist/json_formatter.exe
+
+mkdir bindist/lang
+cp -r lang/mo bindist/lang
+
+$extras = "data", "doc", "gfx", "LICENSE.txt", "LICENSE-OFL-Terminus-Font.txt", "README.md", "VERSION.txt"
+ForEach ($extra in $extras) {
+	cp -r $extra bindist
+}
```
Comment (jbytheway) [inline build-scripts/windist.ps1:13]: I would probably put the `Compress-Archive` step inside this script rather than in the workflow.

**Comment by jbytheway** [comment]
Thanks for working on this.

Note for reviewers: this includes the changes from #51693.

File: .github/workflows/msvc-full-features.yml [hunk 2]
```diff
@@ -95,7 +91,28 @@ jobs:
...
+          msgfmt -f -o ./data/mods/TEST_DATA/lang/mo/ru/LC_MESSAGES/TEST_DATA.mo ./data/mods/TEST_DATA/lang/po/ru.po
+    - name: Run tests
+      run: |
+          .\Cataclysm-test-vcpkg-static-Release-x64.exe --min-duration 0.2 --rng-seed time
+    - name: Assemble windist
+      run: |
+          .\build-scripts\windist.ps1
+    # Although zipping first results in a double zip, we have to do this to
+    # avoid being ratelimited on upload because of the thousands of files
+    # in the dist bundle. This results in a zipped zip, but oh well.
```
[the remark this answers is not in the document]
Comment (akrieger) [reply .github/workflows/msvc-full-features.yml:107]: I mostly wanted to get in something that worked at all, and be able to click and see that this is sane-ish, and try cleaning up the double zip situation later. I didn’t realize there’s a separate release.yml though, I’ll check that out.

File: .github/workflows/msvc-full-features.yml [hunk 3]
```diff
@@ -95,7 +91,28 @@ jobs:
...
+    - name: Compress windist
+      run: |
+          Compress-Archive -Path bindist/* -DestinationPath cataclysmdda-0.F
```
[the remark this answers is not in the document]
Comment (akrieger) [reply .github/workflows/msvc-full-features.yml:110]: It does not. It’s documented that using the glob pattern sets the root as inside the folder. Also when I tested locally it was correct.

File: build-scripts/windist.ps1 (hunk 1 above)
Comment (akrieger) [reply build-scripts/windist.ps1:13]: I had that at first, until I read about the upload-artifacts double zipping issue, so I wanted to leave option value open. But now that I know about release.yml I will look there instead and it’ll probably compress in this script again.

File: build-scripts/windist.ps1 [hunk 4]
```diff
@@ -0,0 +1,13 @@
+rm -Recurse -Force bindist
+
+mkdir bindist
+cp Cataclysm-vcpkg-static-Release-x64.exe bindist/cataclysm-tiles.exe
+cp tools/format/JsonFormatter-vcpkg-static-Release-x64.exe bindist/json_formatter.exe
+
+mkdir bindist/lang
+cp -r lang/mo bindist/lang
+
+$extras = "data", "doc", "gfx", "LICENSE.txt", "LICENSE-OFL-Terminus-Font.txt", "README.md", "VERSION.txt"
```
Comment (BrettDong) [inline build-scripts/windist.ps1:10]: I suggest excluding `data/mods/TEST_DATA` from release distribution.

Comment (akrieger) [reply build-scripts/windist.ps1:10]: That's a good idea, but the official distro also includes this data and for now at least I want to try to do as faithful of an adaptation as possible.

**Comment by akrieger** [comment]
- No longer has someone else's changes misappropriated into it.
- Updated `release.yml` to build and ship cata.
- Fixed VS solution to properly embed manifest and thus have pretty icon on the final binary.
- Changed the VS solution to emit binaries of a consistent name for simplicity sake.

File: build-scripts/windist.ps1 (hunk 4 above)
Comment (akrieger) [reply build-scripts/windist.ps1:12]: I'm initially trying to just match the existing distribution one for one, which does not remove the test data.

File: msvc-full-features/JsonFormatter-vcpkg-static.vcxproj [hunk 5]
```diff
@@ -125,4 +137,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
+</Project>
```
Comment (ZhilkinSerg) [inline msvc-full-features/JsonFormatter-vcpkg-static.vcxproj:140]:
```suggestion
</Project>

```

File: msvc-full-features/Cataclysm-vcpkg-static.vcxproj [hunk 6]
```diff
@@ -117,6 +133,9 @@
       <Project>{0009bb11-11ad-4c14-a5fc-d882a942c00b}</Project>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <ResourceCompile Include="..\src\resource.rc" />
+  </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets" />
-</Project>
+</Project>
```
Comment (BrettDong) [inline msvc-full-features/Cataclysm-vcpkg-static.vcxproj:141]:
```suggestion
</Project>

```

**Comment by akrieger** [comment]
- Restored mxe builds to original state, added msvc builds with `-msvc` added to artifact name.
- Addressed other feedback.

**Comment by akrieger** [comment]
https://github.com/akrieger/Cataclysm-DDA/releases/tag/cdda-experimental-2021-09-30-0020 shows successful builds and I can run them.

**Review by jbytheway — APPROVED** [review]
This looks good to me.

**Pull request merged**